component/portal/src/main/java/org/exoplatform/portal/rest/GroupRestResourcesV1.java

```diff
@@ -641,19 +646,25 @@ public Response getGroupsTreeMembers(@Context
 
     offset = offset > 0 ? offset : DEFAULT_OFFSET;
     limit = limit > 0 ? limit : DEFAULT_LIMIT;
-
-    Collection<Group> UserGroupsList = organizationService.getGroupHandler()
-                                                          .findGroupsOfUserByKeyword(identity.getUserId(), q, groupType);
-    int totalSize = UserGroupsList.size();
+    Group[] groups = null;
+    int totalSize = 0;
+    List<String> excludedGroupsTypes = new ArrayList<>(List.of(SPACES_GROUP_TYPE,ROOT_GROUP_TYPE));
```

Comment:
This list has to be transmitted from REST parameters, even better, this list has to be a list of groups `excludeParentGroup=/space&excludeParentGroup=/` which has to be transformed in Service layer into PLIDM group type and not in REST Layer. In fact, the REST & Service APIs has to be independent from Storage layer that uses for now PLIDM. When we want to change the implementation of Storage layer, we will have to consider changing Service API & REST layers which shouldn't be the case.

Reply:
Updated as requested